apps/meteor/client/views/room/composer/ComposerMessage.tsx

```diff
@@ -85,11 +86,31 @@ const ComposerMessage = ({ tmid, onSend, ...props }: ComposerMessageProps): Reac
 		useCallback(() => LegacyRoomManager.getOpenedRoomByRid(room._id)?.streamActive ?? false, [room._id]),
 	);
 
-	if (!publicationReady) {
+	if (!publicationReady && hasPreviewPublicRoomPermission) {
```

Comment:
What if I'm part of the room and I don't have this permission? Won't it break the loading composer?

Reply:
No, because publicationReady turns into `true`!
But if we proceed with the new design, the one without composer, we should be fine and remove this permission check!